### PR TITLE
feat(skeleton): tokenize forge skeleton

### PR DESCRIPTION
**Pull request by tyl-paul-lapczynski**
## PR Checklist

Please check if your PR fulfills the following requirements:

- Tests for the changes have been added/updated: N
- Docs have been added/updated: N
- Does this PR introduce a breaking change? N
- I have linked any related GitHub issues to be closed when this PR is merged? [Y/N]

## Describe the new behavior?
Uses tokens for styles

## Additional information
This component used special light/dark themed colors previously rgba(255,255,255, 0.3), rgba(255,255,255, 0.12) respectfully. There is no global theme that matches this 100%. Currently left the _mixins and _variables as is for those and looking for guidance for the right decision.

### CODE REVIEW

File: src/lib/skeleton/_core.scss [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    display: block;
+    position: absolute;
+    width: 100%;
+    height: 100%;
+    transform: translateX(-100%);
+    background-repeat: no-repeat;
+    background-image: linear-gradient(
+        90deg,
+        transparent,
+        var(--forge-skeleton-theme-gradient, variables.$gradient),
```
Comment (tyl-paul-lapczynski) [inline src/lib/skeleton/_core.scss:33]: Calling this out as the culprit (see additional information)

File: src/lib/theme/_theme-dark.scss [hunk 2]
```diff
@@ -21,19 +21,17 @@
...
   // Tooltip
-  @include tooltip.provide-theme((
-    background: theme.variable(surface-bright),
-    color: theme.variable(text-high),
-    border-width: border.variable(thin) // Tooltips use thin borders in dark theme to increase contrast
-  ));
+  @include tooltip.provide-theme((background: theme.variable(surface-bright),
+      color: theme.variable(text-high),
+      border-width: border.variable(thin) // Tooltips use thin borders in dark theme to increase contrast
+    ));
```
Comment (DRiFTy17) [inline src/lib/theme/_theme-dark.scss:31]: Let's undo the formatting changes here to reduce the noise.

File: src/lib/skeleton/_variables.scss
Comment (DRiFTy17) [inline src/lib/skeleton/_variables.scss:1]: We can remove this file. Just put the values either in `_tokens.scss` or leave them hardcoded for now and we'll figure out what to do with the theme colors after.

File: src/lib/skeleton/_mixins.scss
Comment (DRiFTy17) [inline src/lib/skeleton/_mixins.scss:1]: We can remove this file. `provide-theme` now exists in your `_token-utils.scss` file that is forwarded from `index.scss`

File: src/lib/skeleton/skeleton.scss [hunk 3]
```diff
@@ -1,13 +1,53 @@
-@use './mixins';
-
-@include mixins.core-styles;
+@use './core' as *;
+@use '../core/styles/theme';
```
Comment (DRiFTy17) [inline src/lib/skeleton/skeleton.scss:2]: I think you can remove this

File: src/lib/skeleton/_variables.scss [hunk 4]
```diff
@@ -1,40 +1,7 @@
...
-  background: $background,
-  gradient: $gradient
-) !default;
+  gradient: $gradient) !default;
 
 $theme-values-dark: (
-  gradient: rgba(255, 255, 255, 0.12),
-  background: rgba(255, 255, 255, 0.12)
-) !default;
+  gradient: rgba(255, 255, 255, 0.12)) !default;
```
Comment (DRiFTy17) [inline src/lib/skeleton/_variables.scss:7]: Move this directly to the `_theme-dark.scss` file, we'll figure out the theming after

File: src/lib/skeleton/_variables.scss [hunk 5]
```diff
@@ -1,40 +1,7 @@
-$background: rgba(0, 0, 0, 0.12) !default;
-$background-color-dark: rgba(255, 255, 255, 0.12) !default;
-$border-radius: 4px !default;
 $gradient: rgba(255, 255, 255, 0.3) !default;
```
Comment (DRiFTy17) [inline src/lib/skeleton/_variables.scss:1]: Move this to `_tokens.scss` for the `gradient-color` token.

File: src/lib/skeleton/_core.scss
Comment (DRiFTy17) [inline src/lib/skeleton/_core.scss:1]: Can you format this file to use 2 spaces? We'll introduce a formatter when we release 3.0, but just wanted to keep it consistent for now so it doesn't change every time someone comes into this file.

File: src/lib/skeleton/skeleton.ts [hunk 6]
```diff
@@ -5,7 +5,38 @@ import { SKELETON_CONSTANTS } from './skeleton-constants';
 import template from './skeleton.html';
 import styles from './skeleton.scss';
 
-export interface ISkeletonComponent extends IBaseComponent {}
+/**
+ * @summary Skeleton is used to provide a placeholder for content that is loading.
```
Comment (DRiFTy17) [inline src/lib/skeleton/skeleton.ts:9]: This needs to be on the `class` below. You can merge it with that, but remove the `"The web component class behind the `<forge-skeleton>` custom element."` text when you do.

**Review by DRiFTy17 — COMMENTED** [review]
Looks good overall. I haven't run the code yet, but just wanted to give you an initial code review.

File: src/lib/skeleton/_core.scss [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+@mixin form-field() {
+    @include override(height, form-field-height);
+    @include override(width, form-field-width);
+}
+
+@mixin chip() {
+    @include override(height, chip-height);
+    @include override(width, chip-width);
+
+    border-radius: 16px;
```
Comment (DRiFTy17) [inline src/lib/skeleton/_core.scss:69]: This should use a global shape token.

**Review by DRiFTy17 — COMMENTED** [review]
After pulling the changes down I noticed that the border radius was missing on a few these elements. Those should come from the global shape tokens. Compare the look of this to the 2.x for parity.

**Comment by tyl-paul-lapczynski** [comment]
Added a new line to files that didn't have one to keep with the standards.

**Pull request merged**